gdal/frmts/grib/gribdataset.cpp

```diff
@@ -826,9 +838,17 @@ CPLErr GRIBRasterBand::IReadBlock( int /* nBlockXOff */,
 
     const int nCopyWords = std::min(nRasterXSize, nGribDataXSize);
 
+    // Split&Swap: transparent rewrap around the prime meridian instead of the antimeridian
```

Comment:
The case where nRasterXSize != nGribDataXSize is a messy one currently (the driver truncates the exposed raster. it should rather use a subdataset mechanism), and I've the feeling that split&swap strategy will produce complete garbage. If we can detect in the SetGribMetaData() function that nRasterXSize != nGribDataXSize, we should disable the split&swap strategy

Reply:
Yes, I was wondering about that comparison